src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -72,11 +72,7 @@ trait Contexts { self: Analyzer =>
 
   lazy val rootImportPackages: List[Symbol] = {
     val paths =
-      if (settings.Ypredef.isSetByUser) {
-        if (settings.nopredef) error("Deprecated flag -Yno-predef isn't supported with -Ypredef")
-        if (settings.noimports) error("Deprecated flag -Yno-imports isn't supported with -Ypredef")
-        settings.Ypredef.value
-      }
+      if (settings.Ypredef.isSetByUser) settings.Ypredef.value
       else if (settings.noimports) Nil
       else if (settings.nopredef) settings.Ypredef.value.filterNot(_ == "scala.Predef")
       else settings.Ypredef.value
```

Comment:
You may not want to go this way, but I was thinking this could be `val paths = settings.Ypredef.value` if setting noimports or nopredef updated Ypredef in the hook. Then the real code doesn't care how Ypredef was composed.